internal/cephfs/volume.go

```diff
@@ -78,12 +90,56 @@ func getVolumeRootPathCeph(ctx context.Context, volOptions *volumeOptions, cr *u
 	return strings.TrimSuffix(stdout, "\n"), nil
 }
 
+func getSubVolumeInfo(ctx context.Context, volOptions *volumeOptions, cr *util.Credentials, volID volumeID) (Subvolume, error) {
+	// keyPresent checks whether corresponding clusterID key is present in clusterAdditionalInfo
+	var keyPresent bool
+	// verify if corresponding ClusterID key is present in the map,
+	// and if not, initialize with default values(false).
+	if _, keyPresent = clusterAdditionalInfo[volOptions.ClusterID]; !keyPresent {
+		clusterAdditionalInfo[volOptions.ClusterID] = &localClusterState{}
+	}
+	info := Subvolume{}
+	err := execCommandJSON(
+		ctx,
+		&info,
+		"ceph",
+		"fs",
+		"subvolume",
+		"info",
+		volOptions.FsName,
+		string(volID),
+		"--group_name",
+		volOptions.SubvolumeGroup,
+		"-m", volOptions.Monitors,
+		"-c", util.CephConfigPath,
+		"-n", cephEntityClientPrefix+cr.ID,
+		"--keyfile="+cr.KeyFile)
+	if err != nil {
+		klog.Errorf(util.Log(ctx, "failed to get subvolume info for the vol %s(%s)"), string(volID), err)
+		if strings.HasPrefix(err.Error(), ErrNotFoundString.Error()) {
+			return info, ErrVolumeNotFound
+		}
+		// Incase the error is other than invalid command return error to the caller.
+		if !strings.Contains(err.Error(), ErrInvalidCommand.Error()) {
+			clusterAdditionalInfo[volOptions.ClusterID].subVolumeInfoSupported = false
+			return info, ErrInvalidCommand
+		}
+
+		return info, err
+	}
+	clusterAdditionalInfo[volOptions.ClusterID].subVolumeInfoSupported = true
```

Comment:
is this value used here where?

Reply:
we could use this or Error return in CreateSnapshot. I opted for Error return, but could use this flag too.  Documented it in the CreateSnapshot() for clarity.